File: Backend/TrackrAPI/TrackrAPI/Dtos/GestionEntidad/EntidadTablaRegistroDto.cs
using System.Collections.Generic;

namespace TrackrAPI.Dtos.GestionEntidad
{
    public class EntidadTablaRegistroDto
    {
        public int Numero { get; set; }
        public int IdEntidadEstructura { get; set; } // Sección
        public int IdTabla { get; set; }
        public List<TablaValorDto> Valores { get; set; } = new List<TablaValorDto>();
    }

    public class TablaValorDto
    {
        public int IdEntidadEstructuraTablaValor { get; set; }
        public string ClaveCampo { get; set; } = string.Empty;
        public string Valor { get; set; } = string.Empty;
        public bool FueraDeRango { get; set; }
        public DateTime FechaMuestra { get; set; }
    }
}

File: Frontend/trackr-administrador/src/app/shared/dtos/gestion-entidades/entidad-tabla-registro-dto.ts
export interface EntidadTablaRegistroDto {
    numero: number;
    idEntidadEstructura: number; // PestanaSección
    idTabla: number;
    valores: TablaValorDto[];
}

export interface TablaValorDto {
    idEntidadEstructuraTablaValor: number;
    idSeccionVariable: number;
    valor: string;
    fueraDeRango: boolean;
    fechaMuestra: Date;
}
